sources/packages/api/src/route-controllers/education-program/education-program.controller.ts

```diff
@@ -146,14 +161,18 @@ export class EducationProgramController {
 
     // The payload returned from form.io contains the approvalStatus as
     // a calculated server value. If the approvalStatus value is sent
-    // from the client form it will be overrided by the server calculated one.
+    // from the client form it will be overridden by the server calculated one.
     const saveProgramPaylod: SaveEducationProgram = {
       ...submissionResult.data.data,
       programDeliveryTypes: submissionResult.data.data.programDeliveryTypes,
       institutionId: userToken.authorizations.institutionId,
       id: programId,
+      userId: userToken.userId,
     };
-    return this.programService.saveEducationProgram(saveProgramPaylod);
+    return this.programService.saveEducationProgram(
+      saveProgramPaylod,
+      createProgram,
```

Comment:
The _createProgram_ can also be define internally in the method checking if the id is present, it would make the code cleaner, IMO. _Typeorm_ is already strongly relying on it to determine insert/update.